Extensions/classic_tags.js

```diff
@@ -63,39 +58,44 @@ XKit.extensions.classic_tags = new Object({
 			value: false
 		}
 	},
+	typeahead_dropdown: null,
+	tag_text: null,
+	tags: [],
+	placeholder: null,
+	search_input: null,
 
 	observer: new MutationObserver(function(mutations) {
-		$("#popover_search .result_link").each(function() {
-			var link = $(this);
-			if (XKit.extensions.classic_tags.preferences.open_in_new_tab.value) {
-				link.attr("target", "_BLANK");
-			} else {
-				link.attr("target", "");
-			}
+		const {classic_tags} = XKit.extensions;
+		const new_tab = classic_tags.preferences.open_in_new_tab.value;
 
-			if (XKit.extensions.classic_tags.preferences.redirect_to_tagged.value) {
-				XKit.extensions.classic_tags.redirect_to_tagged.call(this);
+		mutations.forEach(({addedNodes})=> {
+			if (!addedNodes) {
+				return;
 			}
 
-			var tag_name = link.attr("data-tag-result");
-			var count = XKit.extensions.classic_tags.tagcounts[tag_name];
-			if (count) {
-				var title = link.find(".result_title");
-				title.text(title.text() + " (" + count + ")");
-			}
-		});
-	}),
+			addedNodes.forEach(addedNode => {
+				const container = $(addedNode).filter(classic_tags.typeahead_dropdown);
 
-	redirect_to_tagged: function() {
-		// Extract tag from the data-tag-result attribute. May break if
-		// Tumblr removes this. Will not break if tumblr changes the
-		// link format.
-		var tag = $(this).attr("data-tag-result");
+				if (container.length === 0) {
+					return;
+				}
 
-		// Construct a URL for the tag, replacing all spaces with "-"
-		var newHref = "/tagged/" + tag.replace(/ /g, "-");
-		$(this).attr("href", newHref);
-	},
+				const $tags = container.find("a");
+
+				$tags.each(function() {
+					const $t = $(this);
+
+					$t.attr("target", new_tab ? "_blank" : "");
+
+					const $name = $t.find(classic_tags.tag_text);
```

Comment:
probably nicer to have a not-one-letter variable name

```suggestion
					const $tag = $(this);

					$tag.attr("target", new_tab ? "_blank" : "");

					const $name = $tag.find(classic_tags.tag_text);
```